### PR TITLE
include Flint series in global series()

### PR DESCRIPTION
**Pull request by rwst**
This calls Flint series (if installed) with univariate, else Piranha series (if installed).

### CODE REVIEW

File: symengine/series.cpp [hunk 1]
```diff
@@ -24,9 +29,27 @@ RCP<const SeriesCoeffInterface> series(const RCP<const Basic> &ex, const RCP<con
...
 
+    if (syms.size() == 1)
+#   ifdef HAVE_SYMENGINE_FLINT
+        return URatPSeriesFlint::series(ex, var->get_name(), prec);
+#   else
+        return URatPSeriesPiranha::series(ex, var->get_name(), prec);
+#   endif
+
     return UPSeriesPiranha::series(ex, var->get_name(), prec);
 #else
```
Comment (isuruf) [inline symengine/series.cpp]: You can use `#elif` here. Will be more readable.

**Comment by isuruf** [comment]
Left one comment.
Otherwise looks good to me.

**Comment by isuruf** [comment]
+1 to merge if tests pass

**Comment by rwst** [comment]
Cannot confirm the error, must be Travis-specific.

File: symengine/series.cpp [hunk 2]
```diff
@@ -24,9 +29,25 @@ RCP<const SeriesCoeffInterface> series(const RCP<const Basic> &ex, const RCP<con
...
 
+    if (syms.size() == 1)
+#   ifdef HAVE_SYMENGINE_FLINT
+        return URatPSeriesFlint::series(ex, var->get_name(), prec);
+#   else
+        return URatPSeriesPiranha::series(ex, var->get_name(), prec);
+#   endif
+
     return UPSeriesPiranha::series(ex, var->get_name(), prec);
+#elif HAVE_SYMENGINE_FLINT
```
Comment (isuruf) [inline symengine/series.cpp]: Error is due to this line.
`#elif defined(HAVE_SYMENGINE_FLINT)` should work

File: symengine/series.cpp [hunk 3]
```diff
@@ -24,9 +29,25 @@ RCP<const SeriesCoeffInterface> series(const RCP<const Basic> &ex, const RCP<con
...
+        return URatPSeriesPiranha::series(ex, var->get_name(), prec);
+#   endif
+
     return UPSeriesPiranha::series(ex, var->get_name(), prec);
+#elif defined(HAVE_SYMENGINE_FLINT)
+//! TODO: call generic impl. where needed
+    if (prec == 0)
+        return URatPSeriesFlint::series(integer(0), var->get_name(), prec);
+
+    if (syms.size > 1)
```
Comment (isuruf) [inline symengine/series.cpp]: This should be `syms.size()`

**Comment by rwst** [comment]
Interesting. I would liked to have gotten the error when I compiled locally. Can't we have this synced?

**Comment by isuruf** [comment]
@rwst, you would have had to compile with `WITH_PIRANHA=no WITH_FLINT=yes` to get the compile error. 
> Can't we have this synced?

Sorry, I don't understand what this means

**Comment by rwst** [comment]
Ah I thought my g++ flags were different...

**Comment by isuruf** [comment]
Is the failure because of recent changes in Piranha?

**Comment by rwst** [comment]
Apparently something was fixed in Piranha that hided this bug in the series code.

**Comment by rwst** [comment]
Since the last patch that you didn't comment on is only a tiny change, and everything passes tests now, I'll merge.

**Pull request merged**